puremagic/main.py

```diff
@@ -151,20 +151,35 @@ def _identify_all(header: bytes, footer: bytes, ext=None) -> List[PureMagicWithC
     for matched in matches:
         if matched.byte_match in multi_part_header_dict:
             for magic_row in multi_part_header_dict[matched.byte_match]:
-                start = magic_row.offset
-                end = magic_row.offset + len(magic_row.byte_match)
-                if end > len(header):
-                    continue
-                if header[start:end] == magic_row.byte_match:
-                    new_matches.add(
-                        PureMagic(
-                            byte_match=header[matched.offset : end],
-                            offset=magic_row.offset,
-                            extension=magic_row.extension,
-                            mime_type=magic_row.mime_type,
-                            name=magic_row.name,
+                if "-" in str(magic_row.offset):
+                    start = magic_row.offset
+                    end = magic_row.offset + len(magic_row.byte_match)
+                    match_area = footer[start:end] if end != 0 else footer[start:]
+                    if match_area == magic_row.byte_match:
+                        new_matches.add(
+                            PureMagic(
+                                byte_match=matched.byte_match + magic_row.byte_match,
+                                offset=magic_row.offset,
+                                extension=magic_row.extension,
+                                mime_type=magic_row.mime_type,
+                                name=magic_row.name,
+                            )
+                        )
+                else:
+                    start = magic_row.offset
+                    end = magic_row.offset + len(magic_row.byte_match)
+                    if end > len(header):
```

Comment:
```suggestion
                start = magic_row.offset
                end = magic_row.offset + len(magic_row.byte_match)
                if magic_row.offset < 0:
                    match_area = footer[start:end] if end != 0 else footer[start:]
                    if match_area == magic_row.byte_match:
                        new_matches.add(
                            PureMagic(
                                byte_match=matched.byte_match + magic_row.byte_match,
                                offset=magic_row.offset,
                                extension=magic_row.extension,
                                mime_type=magic_row.mime_type,
                                name=magic_row.name,
                            )
                        )
                else:
                    if end > len(header):
```

Haven't verified the logic / tested myself but just wanted to provide a bit of python specific cleanup. Moving the start and end outside the if statements as they are the same, and check `magic_row.offset < 0` instead of against a string (if that's a problem for some reason let me know.)

Reply:
@cdgriffith, Tested and no issues, I have merged it into the PR. Thanks for the suggestions, my coding skills are ok but I know there's always room for improvement. 🙂